plugins/ocp/ocp-nvme.c

```diff
@@ -828,6 +828,434 @@ static int eol_plp_failure_mode(int argc, char **argv, struct command *cmd,
+				if (output < 0) {
+					err = -13;
+					goto end;
+				}
+			}
+			if (write(output, data, transfersize) < 0) {
+				err = -10;
+				goto close_output;
+			}
+		} else { //last piece
+			if (write(output, data, last_xfer_size) < 0) {
+				err = -10;
+				goto close_output;
+			}
+		}
+		offset += transfersize;
+		printf("%d%%\r", (i + 1) * 100 / total_loop_cnt);
+	}
+	printf("100%%\nThe log file was saved at \"%s\"\n", filepath);
+
+close_output:
+	close(output);
+
+end:
+	free(data);
+	return err;
+}
+
+static int get_telemetry_dump(struct nvme_dev *dev, char *filename, char *sn,
+		enum TELEMETRY_TYPE tele_type, int data_area, bool header_print)
+{
+	__u32 err = 0, nsid = 0x00000000;
+	__u8 lsp = 0, rae = 0;
+
+	char data[TELEMETRY_TRANSFER_SIZE] = {0,};
+	char *featurename = 0;
+
+	if (tele_type == TELEMETRY_TYPE_HOST_0) {
+		featurename = "Host(0)";
+		lsp = 0;
+		rae = 0;
+		tele_type = TELEMETRY_TYPE_HOST;
+	} else if (tele_type == TELEMETRY_TYPE_HOST_1) {
+		featurename = "Host(1)";
+		lsp = 1;
+		rae = 0;
+		tele_type = TELEMETRY_TYPE_HOST;
+	} else { // TELEMETRY_TYPE_CONTROLLER
+		featurename = "Controller";
+		lsp = 0;
+		rae = 1;
+	}
+
+	// Get Header
+	err = get_telemetry_header(dev, nsid, tele_type, TELEMETRY_HEADER_SIZE,
+			(void *)data, lsp, rae);
+	if (err)
+		return err;
+
+	struct telemetry_initiated_log *logheader =
+			(struct telemetry_initiated_log *)data;
+
+	if (header_print)
+		print_telemetry_header(logheader, tele_type);
+
+	__u64 offset = 0, size = 0;
+
+	switch (data_area) {
+	case 1:
+		offset  = TELEMETRY_HEADER_SIZE;
+		size    = le16_to_cpu(logheader->DataArea1LastBlock);
+		break;
+
+	case 2:
+		offset  = TELEMETRY_HEADER_SIZE
+				+ (le16_to_cpu(logheader->DataArea1LastBlock) * TELEMETRY_BYTE_PER_BLOCK);
+		size    = le16_to_cpu(logheader->DataArea2LastBlock)
+				- le16_to_cpu(logheader->DataArea1LastBlock);
+		break;
+
+	case 3:
+		offset  = TELEMETRY_HEADER_SIZE
+				+ (le16_to_cpu(logheader->DataArea2LastBlock) * TELEMETRY_BYTE_PER_BLOCK);
+		size    = le16_to_cpu(logheader->DataArea3LastBlock)
+				- le16_to_cpu(logheader->DataArea2LastBlock);
+		break;
+
+	default:
+		break;
+	}
+
+	if (size == 0) {
+		printf("Telemetry %s Area %d is empty.\n", featurename, data_area);
+		return err;
+	}
+
+	// Get Data Block
+	char dumpname[FILE_NAME_SIZE] = {0,};
+
+	snprintf(dumpname, FILE_NAME_SIZE,
+					"Telemetry_%s_Area_%d", featurename, data_area);
+	err = extract_dump_get_log(dev, dumpname, filename, sn, size * TELEMETRY_BYTE_PER_BLOCK,
+			TELEMETRY_TRANSFER_SIZE, nsid, tele_type,
+			0, offset, rae);
+
+	return err;
+}
+
+static int ocp_telemetry_log(int argc, char **argv, struct command *cmd,
+			      struct plugin *plugin)
+{
+	struct nvme_dev *dev;
+	int err = 0;
+	const char *desc = "Retrieve and save telemetry log.";
+	const char *type = "Telemetry Type; 'host[Create bit]' or 'controller'";
+	const char *area = "Telemetry Data Area; 1 or 3";
```

Comment:
Could you please let me know the reason to allow only the area 1 or 3 since it is described by the OCP documentation as below so seems the area 2 also should be used as option.
![tel-5](https://user-images.githubusercontent.com/44117581/230632235-78b6250a-2efa-45d8-b747-dd8c12a6cd4a.png)